core/storage/base_model/gae_models.py

```diff
@@ -461,6 +390,17 @@ def put_for_bot(self):
         """Stores the model instance on behalf of a non-human."""
         return super(BaseHumanMaintainedModel, self).put()
 
+    def put_depending_on_id(self, user_id):
```

Comment:
Why not just always require using this function to remove the ambiguity of when to choose put_for_bot vs. put_for_human?